### DISCUSSION TITLE
Can't reach database Error

### DISCUSSION THREAD

**Discussion by akarpov91** [supabase/supabase#4161] [Questions] [answered]
Hi there,

I’m building a prototype NextJs app using Prisma ORM and Supabase PostgresSQL database.

However, I’m receiving the following error:

```
Server Error
Error: 
Invalid `prisma.course.create()` invocation:


  Can't reach database server at `db.wzlsuclrgwfuyuedoehj.supabase.co`:`5432`

Please make sure your database server is running at `db.wzlsuclrgwfuyuedoehj.supabase.co`:`5432`.
```

Details:

M1 MacOS Monteray 12.0.1 

Next.js v12.0.4

prisma : 3.5.0

```
generator client {
    provider = "prisma-client-js"
}

datasource db {
    provider = "postgresql"
    url      = env("DATABASE_URL")
}

DATABASE_URL="postgresql://postgres:[password]@db.wzlsuclrgwfuyuedoehj.supabase.co:5432/postgres”
```

Any assistence would be greatly appreciated.

**Comment by ericmarked** [comment]
same

**Comment by ltanady** [comment]
Can you send your project references to support@supabase.io? We'll take a look at this.

**Comment by silentworks** [comment, marked as answer]
Can you make sure your password doesn't have any weird characters in it like a `@` for instance? Also try connecting with a Postgres GUI client from your computer, something like TablePlus or BeeKeeper Studio.

**Reply by Arucard11** [reply]
just restart the server that fixed it for me 
Error: P1001: Can't reach database server at `aws-0-us-east-2.pooler.supabase.com:5432`

**Comment by nimeshmaharjan1** [comment]
It was working then suddenly I got this error but I have not changed the password.

**Reply by MrJadaml** [reply]
Yup, I keep having the same issue on multiple test projects I've spun up. Also even swapped out to a new Supabase project, which worked at first then also suddenly stopped working.

https://vercel.com/guides/nextjs-prisma-postgres

**Reply by devknight216** [reply]
That's right. I am having same issues while using Prisma and Supabase. Still not sure what the reason is. Seems they are still unstable, not ready to be used by community.

**Reply by gijs-ve** [reply]
I get similar issues, sometimes its working fine but at other times I receive the connection error again.

**Reply by pileroniel** [reply]
Same , I'm also facing the same issue with prisma and supabase. And I have to restart the server again for it to work

**Comment by NikolaosArgy** [comment]
I am building a prototype and I am having the same problem. Everything worked fine and suddenly it stopped. I changed the password but still I can't connect. The other project works fine (I have two projects on Supabase). I tested both projects on TablePlus and DBeaver. One project connected normally, but the other did not, both with the same error.
<img width="524" alt="Screenshot 2023-04-27 at 16 24 34" src="https://user-images.githubusercontent.com/41186400/234899266-d1605839-afe6-4e30-a95b-672dcf081074.png">

weirdly Prisma Studio works fine, but Apollo GraphQL Studio does not.

**Reply by CloudInnovator** [reply]
Look at the appropriate connection pooling to use. Note that all Posgres SQL features only work with Session pooling setting  See this link for more details https://supabase.com/docs/guides/database/connecting-to-postgres#how-connection-pooling-works

**Comment by rayli09** [comment]
same on this

**Reply by silentworks** [reply]
What does same on this mean in your case? Please be more descriptive when leaving a comment.

**Reply by AurelianSpodarec** [reply]
Use google translate if needed

**Comment by Jordaneisenburger** [comment]
Having the same issue it was working fine all of the sudden it stops working.. I'm pretty sure it has a correlation to the Prisma client as my login form still works

```cli
Invalid `prisma.users.findMany()` invocation:


Can't reach database server at `db.crpvifgwvaqdzzoldkak.supabase.co`:`5432`

Please make sure your database server is running at `db.crpvifgwvaqdzzoldkak.supabase.co`:`5432`.

```

**Reply by Jordaneisenburger** [reply]
I just created a new Supabase project and it works again, no code changes at al besides switching the DB url's in .env

**Comment by darcy-vitacca** [comment]
What fixed it for me was that a free tier project had gone into inactive mode and I just had to restart it, not sure if this will help anyone.

**Comment by pablosr11** [comment]
Suddenly got this too. No changes to DB credentials. Restarted yesterday and it was still working fine.
One day later `Can't reach database server at `db.XXXXXXX.supabase.co`:`5432`` and `connection timeout expired`

Have tried connecting through pgadmin, prisma, and psql

**Reply by pablosr11** [reply]
update: Support came back and the IP I was using at the time had been blacklisted by fail2ban.

**Reply by JayashTripathy** [reply]
Just use this and change your credentials 

`DATABASE_URL="postgres://postgres:__PASSWORD__@db.__YOUR_SUPABASE_PROJECT__.supabase.co:6543/postgres?pgbouncer=true"`

**Reply by KD-kelvn** [reply]
I had the same issue. I solved it by closing the terminal(delete and open new) plus delete the old(current ".next js "folder) then i restarted my application then greenlight

**Reply by p2pskins** [reply]
https://github.com/orgs/supabase/discussions/4161#discussioncomment-7265953

Why does this solve it?

**Reply by samjusaitis** [reply]
@p2pskins refer to this article: https://supabase.com/partners/integrations/prisma

**Reply by ajagatobby** [reply]
This solution solved it ASAP

**Reply by Oribi5** [reply]
This solved it for me instantly

**Reply by RsRsRahul** [reply]
This  worked for me

**Reply by servehub-llc** [reply]
me too!

**Comment by jmilus** [comment]
Having this same issue with my dev environment (localhost). Curiously, my live site, hosted on Vercel and which uses the same database, works just fine.
Problem appeared overnight. Worked fine last night before bed, and now refuses to function with morning coffee. Restarted everything, updated everything...

**Reply by jmilus** [reply]
I should clarify: both my dev (local) build and my live build point to the exact same SupaBase project, so it wouldn't seem to be an issue with the project being inactive, as it works for one but not the other.

**Comment by nimeshmaharjan1** [comment]
I forgot what I did because has been a long time but I think it was just
enabling the project because supabase disables it?

**Reply by jmilus** [reply]
project not disabled because I can still access it via a published version of my app.

**Reply by nimeshmaharjan1** [reply]
https://github.com/orgs/supabase/discussions/4161#discussioncomment-2262224 what about this?

**Reply by nimeshmaharjan1** [reply]
I remember having a ! In my password in my env file and I changed the password and it worked

**Reply by jmilus** [reply]
I never changed my password from the original that Supabase generated. I just re-generated a new one (using Suapbase's generator) and I'm still getting the same result.

**Reply by jmilus** [reply]
It appears I have solved it. I discovered that the connection string in my database settings (under Connection Pooling) was using a different port from the one in my env file. No idea why that would have changed, as it wasn't anything I would have initiated.

**Reply by devknight216** [reply]
The issues are not related to password at all I think.

**Comment by nimeshmaharjan1** [comment]
Haven't encountered this issue after the first time

**Comment by tylerbecks** [comment]
https://stackoverflow.com/questions/73866587/prisma-error-p1001-cant-reach-database-server-at-db-xocheossqzkirwnhzxxm-sup

This solved it for me. For context, my connection was slow and it was throwing this error intermittently.

**Comment by KuhlTime** [comment]
This is what **Supabase Support** told me: This might happen due to the connection being temporarily blocked by Supabase, see
https://supabase.com/docs/guides/platform/troubleshooting#connection-refused

---

For the time my connection was blocked, I was able to use a VPN (Germany → USA → Supabase) to connect to the instance. After some while, I could then connect normally again.

**Reply by wvovaw** [reply]
I found that I cannot connect my development environment to supabase while connected via VPN. Apparently supabase has blacklisted the IP server where my VPN is hosted.

**Comment by CloudInnovator** [comment]
@Andre This is an interesting find. I am able to connect from Nigeria  🇳🇬
________________________________

**Comment by adnanjpg** [comment]
I am currently experiencing this problem trying to do a migration using `prisma migrate dev`. Depending on some suggestions from the comments, I tried switching my connection to cellular, and tried from an aws ec2 machine running at `us-east-1b` (I'm located at Turkey), but neither fixed it.

But I had no trouble connecting from `TablePlus`.

Apparently, the connection is succeeding, but I had like 10 new tables in the migration. Splitting the migration into 2 fixed it for me

**Comment by oxneat** [comment]
@silentworks @akarpov91 @ltanady @MrJadaml @KuhlTime i think that we have a bigger problem here, that's the password i am using: zK52BvcGlUKAx5kR, and i am still getting this :
![Screenshot from 2023-11-08 14-01-51](https://github.com/supabase/supabase/assets/62510442/11e04f4d-7f59-4ba1-90d9-b8b4ca58660f)
or this:
![Screenshot from 2023-11-08 14-02-06](https://github.com/supabase/supabase/assets/62510442/9ec3698a-b584-4430-b2df-4f55671505f6)

**Comment by max-17** [comment]
If u cant solve the issue just try using alternative like [Neon ](https://neon.tech/) or [PlanetScale](https://planetscale.com/)

**Comment by ss-joy** [comment]
I had the same issue. This is the error message:
 npx prisma migrate dev
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "sy"

Error: P1001: Can't reach database server at `sy`:`5432` 

I tried :
1 Manually creating a table in supabase hoping that might "turn on" the connection in 5432 port.
2 I tried to use the pooling stuff .
But none of that work.

Then I found there was an "@" symbol in my password. So I chaned my password to use only alphabets.
Then i triend 2/3 more times and got the same error.
But finally it connected.


This is the last error I got before successfully connecting.
` npx prisma migrate dev
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "db.hyoisvlrxevwfousqicv.supabase.co:5432 

Error: P1001

Can't reach database server at `db.hyoisvlrxevwfousqicv.supabase.co`:`5432`

Please make sure your database server is running at `db.hyoisvlrxevwfousqicv.supabase.co`:`5432`.`

**Comment by Coderamrin** [comment]
I was getting this error just because the password had numbers in it. 
So make sure to just put alphabets in the db password no numbers no special chars 
Happy coding.

**Reply by tinegaCollins** [reply]
this worked for me

**Comment by MarionbDev** [comment]
Here's the error I found???
In my .env file I have this : 
DATABASE_URL="postgresql://postgres:[my_password]@db.sxcvldvzpvnaynwnugko.supabase.co:5432/postgres"
My password is in alphabets.
and in my PRISMA file :
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql
  url = env("DATABASE_URL")
}

I have, I think, tried everything!!!!

**Reply by auauwolff** [reply]
I have the same issue, have you solved this ?

**Comment by kagawa0710** [comment]
I had the same error and was able to solve it.
In my case, the password contained symbols, so I encoded it as a url and it worked.

**Reply by stevejpurves** [reply]
great solution, this works fine!

**Reply by OdapX** [reply]
this worked for me, isn't this a bug from supabase side ?

**Comment by Ciscodefran** [comment]
Run 'prisma migrate dev' and check if previewFeatures = ["multiSchema"] property needed in your schema.prisma file.

**Comment by angelxtrme** [comment]
Create new password. Use only Letters either in capital or not. This solved this problem.

**Reply by MooJhingjit** [reply]
Thanks, This solved this issue for me

**Comment by adityayaduvanshi** [comment]
Get the Database URI from database settings. it will look like this 
"postgres://postgres.[USER]:[YOUR-PASSWORD]@aws-0-ap-southeast-1.pooler.supabase.com:5432/postgres"

You have to add "connect_timeout=300" in the end of the URI.

This will fix it

**Reply by pwilczewski** [reply]
I suddenly experienced this error this morning - everything had been working last night. For some reason the database connection string in supabase had changed from:

postgresql://postgres:[MY-PASSWORD]@db.[MY-DATABASE].supabase.co:5432/postgres

To: 

postgres://postgres.[MY-DATABASE]:[MY-PASSWORD]@aws-0-us-east-1.pooler.supabase.com:5432/postgres

To my knowledge I hadn't changed any settings in supabase. But updating to the new connection string resolved the error for me.

**Reply by adityayaduvanshi** [reply]
Yeah its because Supabase is migrating to IPV6. 

Use this link and it will fix the error
postgres://postgres.[MY-DATABASE]:[MY-PASSWORD]@aws-0-us-east-1.pooler.supabase.com:5432/postgres?connect_timeout=300

**Reply by AdventureBear** [reply]
this fixed it for me. thanks

**Reply by leofmoura** [reply]
Thanks, spent 2hours trying to fix this. GOLD.

**Reply by tinegaCollins** [reply]
GOLD

**Comment by AdventureBear** [comment]
Ha;ving the same issues. It was working fine last night and this AM getting the same types of errors.  
Ido have symbols in my PW, so I guess I'll try updating the PW.  I restarted the DB server. Seems weird that it happens all of a sudden.

Changed my password and no luck still.

**Reply by AdventureBear** [reply]
fixed it by updating my connection string as show in this thread. 
 it would be great if this could be alerted on the dashboard. I did see the migration alert but mine had been migrated and I didn't realize it except that it stopped working.   

Just a friendly suggestion for a UI improvement.  This cost me about an hour of troubleshooting today (when I should have been working at my real J-O-B, lol).

**Comment by Hiperultimate** [comment]
For people who are recently having this issue I have a fix for it. 

This is the message I found on Supabase dashboard > Project Settings > Database > URI
"Direct database access via IPv4 and pgBouncer will be removed from January 26th 2024
We strongly recommend using connection pooling to connect to your database. You'll only need to change the connection string that you're using in your application to the pooler's connection string."

Supabase have updated something behind the scenes and because of it the connection string is now a bit different. Just use the new one and you will be fine after you add the db password in it.

**Comment by johannesludloff** [comment]
For me the default setting under "Connection pooling configuration" for my db was "Transaction mode", however for all Postgres features to work you need "Session". At least for me it fixed the issue. Also had to change connection string though.

![image](https://github.com/supabase/supabase/assets/63195223/417927ac-79cb-4104-bac0-a324385df334)

**Comment by Eronred** [comment]
After updating the database URL, the issue was fixed. However, receiving an alert whenever the database URL is updated would be helpful

**Comment by frankdavidcorona** [comment]
Hi, I've configured my project using NextJS and Prisma as ORM and my connection strings look like this:

````
# Connect to Supabase via connection pooling with Supavisor.
DATABASE_URL="postgres://postgres.vghrgcxiebaxjgjmhrum:[PASSWORD]@aws-0-us-east-1.pooler.supabase.com:5432/postgres?connect_timeout=300"

# Direct connection to the database. Used for migrations.
DIRECT_URL="postgresql://postgres:[PASSWORD]@db.vghrgcxiebaxjgjmhrum.supabase.co:5432/postgres?connect_timeout=300"
````

Then I tried to run this command to upload my current schemas from Prisma ORM but I still had this connection string error:

````
npx prisma migrate dev --name init
Environment variables loaded from .env
Prisma schema loaded from src/prisma/schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "db.vghrgcxiebaxjgjmhrum.supabase.co:5432"

Error: P1001: Can't reach database server at `db.vghrgcxiebaxjgjmhrum.supabase.co`:`5432`

Please make sure your database server is running at `db.vghrgcxiebaxjgjmhrum.supabase.co`:`5432`.
````

**Comment by henningsieh** [comment]
Same here... since 2 or 3 weeks I think and **in incomprehensible and irregular intervals**. 

```

 ⨯ Error [PrismaClientInitializationError]:
Invalid `prisma.report.findMany()` invocation:

Can't reach database server at `aws-0-eu-central-1.pooler.supabase.com`:`5432`

Please make sure your database server is running at `aws-0-eu-central-1.pooler.supabase.com`:`5432`.
```

**Reply by CloudInnovator** [reply]
> Same here... since 2 or 3 weeks I think and **in incomprehensible and irregular intervals**. 
> 
> 
> 
> ```
> 
> 
> 
>  ⨯ Error [PrismaClientInitializationError]:
> 
> Invalid `prisma.report.findMany()` invocation:
> 
> 
> 
> Can't reach database server at `aws-0-eu-central-1.pooler.supabase.com`:`5432`
> 
> 
> 
> Please make sure your database server is running at `aws-0-eu-central-1.pooler.supabase.com`:`5432`.
> 
> ```

This could be due to a host of reasons associated with connectivity. If you have poor internet connectivity then this error pops up. This is the hardest to debug of all the reasons. I have been blessed with poor internet, that's how I know 😬  Unfortunately, this has more to do with latency associated with all cloud services not supabase. My answer assumes that you have a running instance and followed proper guidance on 
connections, password and pooling.

**Comment by leomerida15** [comment]
> Hi there,
> 
> I’m building a prototype NextJs app using Prisma ORM and Supabase PostgresSQL database.
> 
> However, I’m receiving the following error:
> 
> ```
> Server Error
> Error: 
> Invalid `prisma.course.create()` invocation:
> 
> 
>   Can't reach database server at `db.wzlsuclrgwfuyuedoehj.supabase.co`:`5432`
> 
> Please make sure your database server is running at `db.wzlsuclrgwfuyuedoehj.supabase.co`:`5432`.
> ```
> 
> Details:
> 
> M1 MacOS Monteray 12.0.1
> 
> Next.js v12.0.4
> 
> prisma : 3.5.0
> 
> ```
> generator client {
>     provider = "prisma-client-js"
> }
> 
> datasource db {
>     provider = "postgresql"
>     url      = env("DATABASE_URL")
> }
> 
> DATABASE_URL="postgresql://postgres:[password]@db.wzlsuclrgwfuyuedoehj.supabase.co:5432/postgres”
> ```
> 
> Any assistence would be greatly appreciated.

I managed to solve the error by running the migration script with bunjs, I tested it with several networks using bunjs and it worked on all of them regardless of the internet speed.

> bun prisma migrate dev

**Comment by sashaskull** [comment]
![image](https://github.com/supabase/supabase/assets/13396277/c1a01ff0-22c5-4cdc-9245-9dd8a1e6d707)
Somehow Supabase had banned my IP and it was appearing under _Network Bans_. After removing the said IP it started working straight away.

**Comment by AbdullahAlhariri** [comment]
Quote from supabase:

> Direct database connections will start resolving to an IPv6 address. Starting from January 26, 2024, projects will be gradually migrated to this behavior. If you connect to your database using the REST or GraphQL endpoints, this doesn't affect you. If you use the database connection string, see the GitHub discussion on [how to prepare for IPv6](https://github.com/orgs/supabase/discussions/17817).

I had to follow [how to prepare for IPv6](https://github.com/orgs/supabase/discussions/17817), I only needed to change my connection string. and everything works smoothly now

**Comment by alexshenker** [comment]
When runing `npm run build` in my NextJS (14.2.3) project, random statically generated pages randomly throw this error, its never the same pages that fail to build:
```
PrismaClientKnownRequestError: 
Invalid `prisma.composer.findFirst()` invocation:

Can't reach database server at `aws-0-us-east-1.pooler.supabase.com`:`5432`

Please make sure your database server is running at `aws-0-us-east-1.pooler.supabase.com`:`5432`.
```
I'm using the new connection string `DATABASE_URL=postgres://postgres.[DB ID]:[Password]@aws-0-us-east-1.pooler.supabase.com:5432/postgres`.

I've attempted nearly all of the solutions offered in this thread. None of them worked. The only thing that enabled a successful build, is appending `?connection_limit=1` to the end of my DB string. anything above '1' leads to the same intermittent errors for random pages.

This is not ideal, so any ideas are appreciated.

**Reply by e-Naeim** [reply]
thanks! I've been strugling with it the whole day. I hope someone can explain and provide a solution

**Reply by Sameerthe-Big-O** [reply]
For me guys it was just a type carefully copy and paste the connection string

**Comment by Jmadih** [comment]
When using special characters like &, !, or % in a PostgreSQL URL, they can cause issues because they are interpreted as part of the URL syntax, not as part of the password.

To resolve this, you'll need to URL-encode special characters in your password. URL encoding replaces special characters with their corresponding percent-encoded values. For instance:

& becomes %26
! becomes %21

**Comment by ghost** [comment]
I had the same issue.  Can't reach database server at ...

1. Changed the password. Used just one that the system recommended. (Mine was containing an @)
2. Restarted the project.

All working fine.

**Comment by abayega** [comment]
I ran into this issue using company internet connection. I turned on my phone hotspot and used my cell phone network internet and that solved the problem.
